### DISCUSSION TITLE
Error during install: failed to be installed because End of Central Directory could not be found

### DISCUSSION THREAD

**Discussion by RichSteele** [JanDeDobbeleer/oh-my-posh#555] [General]
Trying to install in PowerShell Core 7.1.3 on Windows 10 1909 but getting an error:

```
Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): y
Install-Package: C:\program files\powershell\7\Modules\PowerShellGet\PSModule.psm1:9711
Line |
9711 |  … talledPackages = PackageManagement\Install-Package @PSBoundParameters
     |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Package 'oh-my-posh' failed to be installed because: End of Central Directory record could not be
     | found.
```

I suspect, though my inexperience with PowerShell is not good enough to tell for sure, that my corporate proxy is the root cause. It does SSL interception, so there's a custom root certificate that frequently causes issues with other tools that download content. Googling finds some similar issues installing (some) other PowerShell modules, though the methods to resolve are lost on me.

Note that I was able to successfully install posh-git (with the same warning about an untrusted repository but no error during the install).

Does anyone else see this? Does anyone have suggestions on how to resolve it?

**Comment by JanDeDobbeleer** [comment]
Probably an issue with PowershellGet. Open an elevated powershell prompt and enter the following commands:

```pwsh
Install-PackageProvider Nuget –force –verbose

Install-Module –Name PowerShellGet –Force –Verbose

Update-Module
```

**Reply by RichSteele** [reply]
I get an error trying to install nuget as a package provider:
```
PS C:\Windows\System32> Install-PackageProvider -Name NuGet -Force -MinimumVersion 2.8.5.201 -Verbose
VERBOSE: Using the provider 'Bootstrap' for searching packages.
VERBOSE: Finding the package 'Bootstrap::FindPackage' 'NuGet','','2.8.5.201','''.
Install-PackageProvider: No match was found for the specified search criteria for the provider 'NuGet'. The package provider requires 'PackageManagement' and 'Provider' tags. Please check if the specified package has the tags.
```
I was able to install PowerShellGet without error. It looks like both are installed:
```
PS C:\Windows\System32> Get-PackageProvider

Name                     Version          DynamicOptions
----                     -------          --------------
NuGet                    3.0.0.1          Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag, …
PowerShellGet            2.2.5.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, In…
```
Finally, I still get the same error when installing oh-my-posh:
```
PS C:\Users\rsteele> Install-Module oh-my-posh -Scope CurrentUser

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): y
Install-Package: C:\Users\rsteele\OneDrive - Paychex\Documents\PowerShell\Modules\PowerShellGet\2.2.5\PSModule.psm1:9711
Line |
9711 |  … talledPackages = PackageManagement\Install-Package @PSBoundParameters
     |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Package 'oh-my-posh' failed to be installed because: End of Central Directory record could not be
     | found.
```

**Reply by JanDeDobbeleer** [reply]
Ok, in that case it's definitely going to be the corporate network given that AD is mentioned and the install fails (all of these are things we don't control, the error is coming from pwsh itself).

**Reply by RichSteele** [reply]
I think the reference to "central directory" is what happens when trying to unzip a corrupted zip file, which is often the result of the kind of SSL interception these corporate proxies do.

So, is this something to be brought up with the PowerShell project then?

**Reply by RichSteele** [reply]
Annnnnd, never mind. I get an error from our corporate proxy with a "suspect virus" when downloading manually.

Thanks for your help.

**Reply by JanDeDobbeleer** [reply]
Know I have a case running at MS to get exclusion for oh-my-posh when it comes to false positives. It's insane really, but we can't fix that on our end.

**Comment by weyCC81** [comment]
Try to access the Url "https://cdn.powershellgallery.com/packages/oh-my-posh.7.85.2.nupkg" 
or do the following: `Install-Module -Name oh-my-pos -Scope AllUsers -Force -Debug -Verbose`

> VERBOSE: Hash for package ' oh-my-pos' does not match hash provided from the server.

For me, it was the Firewall Blocking the request to the "nupkg".

**Reply by JanDeDobbeleer** [reply]
@weyCC81 be aware that is a very old package that you're installing because we moved away from the PowerShell gallery years ago. Version 8 is really not what you want. I advise you to go look at a website to install the latest version using Winget instead.
